### PR TITLE
doc: add doc about installing apisix ingress with kind

### PR DESCRIPTION
**Pull request by Chever-John**
As title says.

### Type of change:

- [ ] Bugfix
- [ ] New feature provided
- [ ] Improve performance
- [ ] Backport patches

### Pre-submission checklist:

* [x] Did you explain what problem does this PR solve? Or what new features have been added?
* [ ] Have you added corresponding test cases?
* [x] Have you modified the corresponding document?
* [ ] Is this PR backward compatible? **If it is not backward compatible, please discuss on the [mailing list](https://github.com/apache/apisix-ingress-controller#community) first**

### CODE REVIEW

File: docs/en/latest/deployments/kind.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
+---
+title: Install Ingress APISIX on Kind
```
Comment (juzhiyuan) [inline docs/en/latest/deployments/kind.md:2]: How about this?

```suggestion
title: Install Ingress Controller by Kind
```

`by` or `on`?

Comment (tao12345666333) [reply docs/en/latest/deployments/kind.md:2]: on

File: docs/en/latest/deployments/kind.md [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+kind: Cluster
+apiVersion: kind.x-k8s.io/v1alpha4
+nodes:
+- role: control-plane
+  kubeadmConfigPatches:
+  - |
+    kind: InitConfiguration
+    nodeRegistration:
+      kubeletExtraArgs:
+        node-labels: "ingress-ready=true"
```
Comment (tao12345666333) [inline docs/en/latest/deployments/kind.md:48]: IIRC, we don't need these configurations.

Comment (Chever-John) [reply docs/en/latest/deployments/kind.md:48]: Yeah, we just use `kind create cluster`.
And then run as follows:
```shell
helm repo add apisix https://charts.apiseven.com
helm repo add bitnami https://charts.bitnami.com/bitnami
helm repo update
kubectl create ns ingress-apisix
helm install apisix apisix/apisix \
  --set gateway.type=NodePort \
  --set ingress-controller.enabled=true \
  --namespace ingress-apisix \
  --set ingress-controller.config.apisix.serviceNamespace=ingress-apisix
kubectl get service --namespace ingress-apisix
```

Everything are done.

I will delete these configurations.

**Comment by tao12345666333** [comment]
Please fix the lint error.

File: docs/en/latest/deployments/kind.md [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+# limitations under the License.
+#
+-->
+
+This document explains how to install Ingress APISIX on [kind](https://kind.sigs.k8s.io/).
+
+## Prerequisites
+
+* Install [kind](https://kind.sigs.k8s.io/docs/user/quick-start/).
+* Install [Helm](https://helm.sh/).
```
Comment (gxthrj) [inline docs/en/latest/deployments/kind.md:29]: Need to specify the versions of `kind` and `helm`.

Comment (tao12345666333) [reply docs/en/latest/deployments/kind.md:31]: I don't think we need to specify that GO/Docker etc. are needed, they are actually kind dependencies.

We just need to note that you need to properly install kind and use it to create a Kubernetes cluster.

Comment (Chever-John) [reply docs/en/latest/deployments/kind.md:31]: After considering, I agree.

**Review by tao12345666333 — APPROVED** [review]

**Review by gxthrj — APPROVED** [review]

**Pull request merged**